src/providers/data_provider/dp_targets.c

```diff
@@ -290,6 +290,12 @@ static errno_t dp_target_init(struct be_ctx *be_ctx,
          * configured so we shall just continue. */
         DEBUG(SSSDBG_CONF_SETTINGS, "Target [%s] is not supported by "
               "module [%s].\n", target->name, target->module_name);
+
+        /* Target is not initialized in this case so we can free
+         * its resources. However this is not an error so we return EOK. */
+        talloc_free(target->methods);
```

Comment:
I guess having `target->initialized = false` should be enough, but perhaps it still better/safer to use `talloc_zfree()` to avoid dangling pointer?